public-ui/src/app/views/breadcrumb/component.spec.ts

```diff
@@ -0,0 +1,218 @@
+import {Component} from '@angular/core';
+import {async, ComponentFixture, fakeAsync, TestBed, tick} from '@angular/core/testing';
+import {Router, Routes} from '@angular/router';
+import {RouterTestingModule} from '@angular/router/testing';
+
+import {BreadcrumbComponent} from './component';
+
+const testRoutes: Routes = [
+  {
+    path: '',
+    component: Component,
+    data: {title: 'root'},
+    children: [
+      {
+        path: 'params/:p1',
+        component: Component,
+        data: {
+          title: 'params',
+          breadcrumb: {
+            value: ':p1',
+            intermediate: false
+          }
+        }
+      },
+      {
+        path: 'child1',
+        component: Component,
+        data: {
+          title: 'child1',
+          breadcrumb: {
+            value: 'child1',
+            intermediate: false
+          }
+        },
+        children: [
+          {
+            path: 'grandchild',
+            component: Component,
+            data: {
+              title: 'grandchild',
+              breadcrumb: {
+                value: 'grandchild',
+                intermediate: false
+              }
+            }
+          }
+        ]
+      },
+      {
+        path: 'child2',
+        component: Component,
+        data: {
+          title: 'child2',
+          breadcrumb: {
+            value: 'child2',
+            intermediate: false
+          }
+        }
+      },
+      {
+        path: 'workspaceChild',
```

Comment:
These are not relevant to data browser may be.

Reply:
its a string used for testing.